mitosheet/mitosheet/api/get_available_snowflake_options_and_defaults.py

```diff
@@ -61,11 +61,20 @@ def get_available_snowflake_options_and_defaults(params: Dict[str, Any], steps_m
 
         con = con_or_exception
 
+        _role = table_loc_and_warehouse.get('role')
         _warehouse = table_loc_and_warehouse.get('warehouse')
         _database = table_loc_and_warehouse.get('database') 
         _schema = table_loc_and_warehouse.get('schema')
         _table_or_view = table_loc_and_warehouse.get('table_or_view')
 
+        roles = get_roles(con)
+        role = _role if _role is not None else get_default_role(roles)
```

Comment:
Let's not make this go and do another DB lookup _every time_. The performance here is already pretty bad. 

you can just move the `get_roles` call down to the next line, as a `get_default_role` parameter.

Reply:
Okay, unfortunetly this is taking longer than expected, so will merge as is.